pkg/defaultRules/defaultRules.yaml

```diff
@@ -1721,7 +1721,43 @@ rules:
     impact: Having non-root execution integrated at build time provides better assurance that applications will function correctly without root privileges
     schema:
       definitions:
-        containerSecurityPattern:
+        podSecurityPattern:
+          properties:
+            spec:
+              properties:
+                securityContext:
+                  properties:
+                    runAsNonRoot:
+                      const: true
+                  required:
+                    - runAsNonRoot
+              required:
```

Comment:
In general, I suggest putting the `required` key above the `properties` key, and not the other way around, it's easier to read